pkg/monitoring/monitoring.go

```diff
@@ -246,14 +248,62 @@ func (tr *templateRegexp) translate(value string) *templateRegexpResult {
 func GetOverwroteAppAnswers(annotations map[string]string) map[string]string {
 	overwritingAppAnswers := annotations[cattleOverwriteAppAnswersAnnotationKey]
 	if len(overwritingAppAnswers) != 0 {
-		var appOverwriteInput mgmtv3.MonitoringInput
+		var appOverwriteInput map[string]string
 		err := json.Unmarshal([]byte(overwritingAppAnswers), &appOverwriteInput)
 		if err == nil {
-			return appOverwriteInput.Answers
+			return appOverwriteInput
 		}
 
 		logrus.Errorf("failed to parse app overwrite input from %q, %v", overwritingAppAnswers, err)
 	}
 
 	return map[string]string{}
 }
+
+func GetMonitoringCreatorID(anno map[string]string) string {
+	if anno == nil {
+		return ""
+	}
+	creator, ok := GetOverwroteAppAnswers(anno)[cattleCreatorIDAnnotationKey]
+	if ok {
+		return creator
+	}
+	return anno[cattleCreatorIDAnnotationKey]
+}
+
+func ValidateCreator(anno map[string]string, userLister mgmtv3.UserLister) (bool, error) {
+	if anno == nil {
+		return false, nil
+	}
+	creatorID := GetMonitoringCreatorID(anno)
+	user, err := userLister.Get("", creatorID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	if user.Enabled == nil || !*user.Enabled {
```

Comment:
Should not need to check `!*userEnabled` here, as we can `return *user.Enabled, nil` at the bottom.